Tasks/JavaToolInstaller/FileExtractor/JavaFilesExtractor.ts

```
@@ -0,0 +1,149 @@
+    
+            // change the console output code page to UTF-8.
+            let chcpPath = path.join(process.env.windir, "system32", "chcp.com");
+            await taskLib.exec(chcpPath, '65001');
+    
+            // run powershell
+            let powershell: tr.ToolRunner = taskLib.tool('powershell')
+                .line('-NoLogo -Sta -NoProfile -NonInteractive -ExecutionPolicy Unrestricted -Command')
+                .arg(command);
+            await powershell.exec();
+        }
+        else {
+            let unzip: tr.ToolRunner = taskLib.tool('unzip')
+                .arg(file);
+            await unzip.exec(<tr.IExecOptions>{ cwd: destinationFolder });
+        }
+    }
+
+    private async sevenZipExtract(file, destinationFolder) {
+        console.log(taskLib.loc('SevenZipExtractFile', file));
+        var sevenZip = taskLib.tool(this.getSevenZipLocation());
+        sevenZip.arg('x');
+        sevenZip.arg('-o' + destinationFolder);
+        sevenZip.arg(file);
+        var execResult = sevenZip.execSync();
+        if (execResult.code != taskLib.TaskResult.Succeeded) {
+            taskLib.debug('execResult: ' + JSON.stringify(execResult));
+        }
+    }
+
+    private async tarExtract(file, destinationFolder) {
+        console.log(taskLib.loc('TarExtractFile', file));
+        let tr: tr.ToolRunner = taskLib.tool('tar');
+        tr.arg(['xzC', destinationFolder, '-f', file]);
+        tr.exec();
+    }
+
+    private failTask(message: string) {
+        taskLib.debug(message);
+        taskLib.setResult(taskLib.TaskResult.Failed, message);
+    }
+
+    private extractFiles(file: string, fileEnding: string) {
+        var stats = taskLib.stats(file);
+        if (!stats) {
+            this.failTask(taskLib.loc('ExtractNonExistFile', file));
```

I prefer the pattern to throw instead of logging it here. The main block is in a try/catch, so it will handle setting the task result.